packages/next/next-server/server/next-server.ts

```diff
@@ -454,9 +454,12 @@ export default class Server {
             fn: async (_req, res, params, _parsedUrl) => {
               const parsedDestination = parseUrl(route.destination, true)
               const destQuery = parsedDestination.query
-              let destinationCompiler = compilePathToRegex(
-                `${parsedDestination.pathname!}${parsedDestination.hash || ''}`
-              )
+              let destToCompile = `${parsedDestination.pathname!}${parsedDestination.hash ||
+                ''}`
+
+              let destinationCompiler = compilePathToRegex(destToCompile, {
+                validate: false,
```

Comment:
I'm still on the fence about this, but this is required if we want to support `:0`.

Can we only skip validation for `:0` somehow?

Reply:
We can skip validation for `:0` by doing the initial commit [here](https://github.com/zeit/next.js/pull/10291/commits/9b3457560c3306c5254da648adeec8058185e31d) which replaces the patterns instead

Reply:
Note: the mentioned issue is not limited to unnamed regexes but named also. I added a test to reflect this here: https://github.com/zeit/next.js/pull/10291/files#diff-40a2f83c72ec325a3777e5ef9df5b60eR293